src/charm.py

```diff
@@ -98,68 +85,70 @@ def _configure_pod(self):
         self.model.pod.set_spec(spec)
         self.unit.status = ActiveStatus()
 
-    def _build_pod_spec(self):
+    def _build_pod_spec(self) -> dict:
+        """This method builds the pod_spec"""
+        if not self.unit.is_leader():
```

Comment:
This is good. So pod spec is only set by leader. And since `self.env_config` is only used in `_build_pod_spec`, only the leader will set the password in the environment of all the charm units. Until here everything is fine. However Juju can change the leader at any time. Or worse your leader unit may die and a new leader may get elected. So how does the new leader know that the old leader had already set a password ? I think your charm may also need to detect and handle `leader_elected` and `leader_settings_changed` events. In handling these events the old leader may need to tell the new leader the password. May be a good idea to get @jlounder opinion on this too.

Reply:
> Or worse your leader unit may die and a new leader may get elected

I had 2 units and simulated this by running: `microk8s.kubectl delete pod mysql-0 -n mysql` 
With this I deleted the leader unit. After the unit was deleted, juju automatically created a new leader unit and this unit was created using the `pod_spec` created when I deployed the application with 2 units for the first time, so the new leader unit uses the `MYSQL_ROOT_PASSWORD` generated when the application was deployed.

Certainly I need to write a test for this.